tests/src/test/scala/whisk/core/containerpool/docker/test/DockerClientWithFileAccessTests.scala

```diff
@@ -0,0 +1,196 @@
+
+    /** Returns a DockerClient with mocked results */
+    def dockerClient(
+        execResult: Future[String] = Future.successful(ipInDockerInspect.asString),
+        readResult: Future[JsObject] = Future.successful(dockerConfig)) =
+        new DockerClientWithFileAccess()(global) {
+            override val dockerCmd = Seq(dockerCommand)
+            override def executeProcess(args: String*)(implicit ec: ExecutionContext) = execResult
+            override def configFileContents(configFile: File) = readResult
+            // Make protected ipAddressFromFile available for testing - requires reflectiveCalls
+            def publicIpAddressFromFile(id: ContainerId, network: String): Future[ContainerIp] = ipAddressFromFile(id, network)
+        }
+
+    behavior of "DockerClientWithFileAccess - ipAddressFromFile"
+
+    it should "throw NoSuchElementException if specified network is not in configuration file" in {
+        val dc = dockerClient()
+
+        a[NoSuchElementException] should be thrownBy await(dc.publicIpAddressFromFile(id, "foo network"))
+    }
+
+    behavior of "DockerClientWithFileAccess - inspectIPAddress"
+
+    it should "read from config file" in {
+        val dc = dockerClient()
+
+        await(dc.inspectIPAddress(id, networkInConfigFile)) shouldBe ipInConfigFile
+        logLines.foreach { _ should not include (s"${dockerCommand} inspect") }
+    }
+
+    it should "fall back to 'docker inspect' if config file cannot be read" in {
+        val dc = dockerClient(readResult = Future.failed(new RuntimeException()))
+
+        await(dc.inspectIPAddress(id, networkInDockerInspect)) shouldBe ipInDockerInspect
+        logLines.head should include(s"${dockerCommand} inspect")
+    }
+
+    it should "throw NoSuchElementException if specified network does not exist" in {
+        val dc = dockerClient(execResult = Future.successful("<no value>"))
+
+        a[NoSuchElementException] should be thrownBy await(dc.inspectIPAddress(id, "foo network"))
+    }
+}
+
+@RunWith(classOf[JUnitRunner])
+class DockerClientWithFileAccessTestsLogs extends FixtureFlatSpec with Matchers with StreamLogging with BeforeAndAfterEach {
+
+    override def beforeEach = stream.reset()
+
+    implicit val transid = TransactionId.testing
+
+    behavior of "DockerClientWithFileAccess - rawContainerLogs"
+
+    /** Returns a DockerClient with mocked results */
+    def dockerClient(logFile: File) = new DockerClientWithFileAccess()(global) {
+        override def containerLogFile(containerId: ContainerId) = logFile
+    }
+
+    def await[A](f: Future[A], timeout: FiniteDuration = 500.milliseconds) = Await.result(f, timeout)
+
+    case class FixtureParam(file: File, writer: FileWriter, dc: DockerClientWithFileAccess)
+
+    def withFixture(test: OneArgTest) = {
+        val file = File.createTempFile(this.getClass.getName, test.name.replaceAll("[^a-zA-Z0-9.-]", "_"))
+        val writer = new FileWriter(file)
+        val dc = dockerClient(file)
```

Comment:
Can we name this `docker`? I have a hard time reading the tests below because it feels like some magic variable to me.